### PR TITLE
Fixing End to End test cases 

### PR DESCRIPTION
**Pull request by mishra14**
Fixes E2E test failure on CI: https://github.com/NuGet/Home/issues/3435
1. Fixing the logic to detect when a if a package is installed in UpdatePackageCommand
2. Fixing the call to Get-VSVersion in VS.ps1 script.

//cc: @joelverhagen @alpaix @emgarten @drewgil @rohit21agrawal @jainaashish @zhili1208 @dtivel

### CODE REVIEW

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 1]
```diff
@@ -183,10 +182,14 @@ private void PerformPackageUpdatesOrReinstalls()
         /// </summary>
         /// <param name="project"></param>
         /// <returns></returns>
-        private async Task PreviewAndExecuteUpdateActionsforSinglePackage()
+        private async Task<bool> PreviewAndExecuteUpdateActionsforSinglePackage()
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Doc comments aren't necessary on non-public members, so either remove or update them.

Comment (dtivel) [reply src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: for -> For

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 2]
```diff
@@ -161,9 +161,8 @@ private void PerformPackageUpdatesOrReinstalls()
         {
             try
             {
-                await PreviewAndExecuteUpdateActionsforSinglePackage();
-
-                if (!_isPackageInstalled)
+                var isPackageInstalled = await PreviewAndExecuteUpdateActionsforSinglePackage();
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: `IsPackageInstalled(...)` returning bool is natural, and call sites read very naturally for it.

In contrast, it doesn't make sense that `PreviewAndExecuteUpdateActionsforSinglePackage()` returns bool; one has to read its implementation to understand what its return value represents.  This problem is solved by moving the call to `IsPackageInstalled(...)` to here, like so:

```
var isPackageInstalled = await IsPackageInstalled(Id);

if (isPackageInstalled)
{
    await PreviewAndExecuteUpdateActionsForSinglePackage();
}
else
{
    Log(MessageLevel.Error, Resources.Cmdlet_PackageNotInstalledInAnyProject, Id);
}
```

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 3]
```diff
@@ -183,10 +182,14 @@ private void PerformPackageUpdatesOrReinstalls()
         /// </summary>
         /// <param name="project"></param>
         /// <returns></returns>
-        private async Task PreviewAndExecuteUpdateActionsforSinglePackage()
+        private async Task<bool> PreviewAndExecuteUpdateActionsforSinglePackage()
         {
             var actions = Enumerable.Empty<NuGetProjectAction>();
 
+            // Check if the package is installed or not.
+            // Used to throw an error message indicating that the package was not installed to begin with.
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: These comments become superfluous with the above changes.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 4]
```diff
@@ -211,33 +214,51 @@ private void PerformPackageUpdatesOrReinstalls()
...
+        /// <summary>
+        /// Method checks if the package to be updated is installed in any package or not.
+        /// </summary>
+        /// <param name="packageId">Id of the package to be updated/checked</param>
+        /// <returns><code>bool</code> indicating wether the package is installed on any package or not</returns>
+        private async Task<bool> IsPackageInstalled(string packageId)
+        {
+            foreach (var project in Projects)
+            {
+                var installedPackages = await project.GetInstalledPackagesAsync(CancellationToken.None);
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: Shouldn't this use [Token](https://github.com/NuGet/NuGet.Client/blob/5741b065cb5c27a9e82f3c67056f46cc80ab550c/src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs#L155), a property inherited from the base class `NuGetPowerShellBaseCommand`?

Side note:  the superclass `PackageActionBaseCommand` also has several locations which use `CancellationToken.None`.  Not sure how many of those should also use `Token`.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 5]
```diff
@@ -211,33 +214,51 @@ private void PerformPackageUpdatesOrReinstalls()
...
             await ExecuteActions(actions);
+
+            return isInstalled;
+        }
+
+        /// <summary>
+        /// Method checks if the package to be updated is installed in any package or not.
+        /// </summary>
+        /// <param name="packageId">Id of the package to be updated/checked</param>
+        /// <returns><code>bool</code> indicating wether the package is installed on any package or not</returns>
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: whether

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 6]
```diff
@@ -211,33 +214,51 @@ private void PerformPackageUpdatesOrReinstalls()
...
-            // set _installed to true, if package to update is installed.
-            _isPackageInstalled = actions.Any();
-
             await ExecuteActions(actions);
+
+            return isInstalled;
+        }
+
+        /// <summary>
+        /// Method checks if the package to be updated is installed in any package or not.
```
Comment (dtivel) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs:221]: ...is already installed or not.

**Comment by dtivel** [comment]
With recommended changes, looks good.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 7]
```diff
@@ -211,33 +214,51 @@ private void PerformPackageUpdatesOrReinstalls()
...
+        /// </summary>
+        /// <param name="packageId">Id of the package to be updated/checked</param>
+        /// <returns><code>bool</code> indicating wether the package is installed on any package or not</returns>
+        private async Task<bool> IsPackageInstalled(string packageId)
```
Comment (alpaix) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs]: `IsPackageInstalledAsync(string packageId, CancellationToken token)`

**Review by joelverhagen — APPROVED** [review]

**Review by alpaix — APPROVED** [review]

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 8]
```diff
@@ -211,33 +214,49 @@ private void PerformPackageUpdatesOrReinstalls()
...
-            // set _installed to true, if package to update is installed.
-            _isPackageInstalled = actions.Any();
-
             await ExecuteActions(actions);
         }
 
         /// <summary>
-        /// Execute the project actions 
+        /// Method checks if the package to be updated is installed in any package or not.
+        /// </summary>
```
Comment (jainaashish) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs:222]: installed in any project or not.

File: src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs [hunk 9]
```diff
@@ -211,33 +214,49 @@ private void PerformPackageUpdatesOrReinstalls()
...
+        /// <param name="packageId">Id of the package to be updated/checked</param>
+        /// <returns><code>bool</code> indicating whether the package is already installed, on any project, or not</returns>
+        private async Task<bool> IsPackageInstalledAsync(string packageId)
+        {
+            foreach (var project in Projects)
+            {
+                var installedPackages = await project.GetInstalledPackagesAsync(Token);
+                if (installedPackages.Select(installedPackage => installedPackage.PackageIdentity.Id)
+                                     .Any(installedPackageId => installedPackageId.Equals(packageId, StringComparison.OrdinalIgnoreCase)))
+                {
```
Comment (jainaashish) [inline src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs:232]: long line, break it.

**Review by jainaashish — APPROVED** [review]
LGTM

**Pull request merged**